### ISSUE TITLE
Fix Missing Export to Retrieve Null Values

### ISSUE THREAD

**Issue by davidclaveau** [bcgov/bcparks-ar-api#377]
#### Description:

Currently, the Missing Export finds "missing" data only if it's also flagged as a `variance` in the database. A record receives the variance flag when valid data is entered for the following year (not left as `null`). Consequently, `null` values aren't considered during this flagging process.

In essence, data is flagged as "missing" under these conditions:
- Valid data existed for the activity in the previous year
- Data entered for the current year is `0`

We therefore want to modify the Missing Data export to detect instances where:
- One month's data is `null`, but
- There was valid data for that activity during the same month last year

This would find missing data outside of just `variance`s, focusing on cases where data exists for a particular month in the previous year but is absent for the same month the next year.

#### Acceptance Criteria:

- [ ] The Missing Data export identifies records where data exists for a specific month in the previous year but is `null` for the same month the next year
- [ ] The export displays relevant information for each identified record, including activity name, month (current year), value from previous year, and `null` indicator for that next year.

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [  Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by manuji** [comment]
Tested on TEST: **Passed**
* Verified that this feature has been implemented. 
* Verified if a field is empty or zero while having data for one or more of previous year for past 3 years, then the flag for missing data is triggered. 
* Passing to PO review. 

@davidclaveau In your previous comment you mentioned that it will check for past 2 years, but when I testing I found out it checks for last 3 years. Check the example given below;
![Image](https://github.com/user-attachments/assets/0d3b5bd7-ccad-4dc5-ba7b-365968ecbe20)

Also I am curious to know what the Notes column contains on Exported report. I checked if it's exporting the value on Enter data page Notes field but it's not.
